nixos/modules/services/networking/radicale.nix

```diff
@@ -3,10 +3,30 @@
 with lib;
 
 let
-
   cfg = config.services.radicale;
 
-  confFile = pkgs.writeText "radicale.conf" cfg.config;
+  format = let
```

Comment:
Doesn't the `ini` format work here? That would be **much** easier to read...

Reply:
No, that's why I had to create my own. The IP addresses Radicale listens on have to be given as a comma-separated list: https://radicale.org/3.0.html#tutorials/basic-configuration/addresses. I thought requiring users to specify e.g. `hosts = "0.0.0.0:5232, [::]:5232"` is a little unnatural.

Reply:
Do we have a (unit) test to ensure our formatter is doing what we expect it to do?

Reply:
Yes, there is `nixos/tests/radicale3.nix`.

Reply:
No no I mean: can we actually format the IPs as we claim? The nixos (integration) test just covers one case as far as I can see. I would love to have a few tests. One case for each of the permissible input values.

Reply:
And essentially, I'm not doing anything except for replacing lists of strings with comma-separated strings and then I use `lib.generators.toINI`, which is tested in `lib/tests/misc.nix`.

Reply:
> No no I mean: can we actually format the IPs as we claim? The nixos (integration) test just covers one case as far as I can see. I would love to have a few tests. One case for each of the permissible input values.

Wouldn't we have to put the generator in `lib/` for that? I honestly don't think unit tests are needed for such a simple thing as `concatStringsSep ", "`.

Reply:
I extended `ini` to support this with https://github.com/NixOS/nixpkgs/pull/121613

Reply:
Thanks a lot!